### PR TITLE
gitg: updated to 3.30.0

### PR DESCRIPTION
**Pull request by albfan**

### CODE REVIEW

**Comment by Alexpux** [comment]
@albfan you need add meson to makedepends

**Comment by albfan** [comment]
Can see details of CI failure, don't know what to solve. Any tip @Alexpux?

**Comment by Alexpux** [comment]
@albfan I'm already told you what to do. Read messages

File: mingw-w64-gitg/PKGBUILD [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# Maintainer: Ignacio Casal Quinteiro <icq@gnome.org>
+# Maintainer: Alberto Fanjul Alonso <albfan@gnome.org>
 
 _realname=gitg
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
-pkgver=3.26.0
+pkgver=3.30.0
 pkgrel=2
```
Comment (lazka) [inline mingw-w64-gitg/PKGBUILD]: pkrel=1

Comment (albfan) [reply mingw-w64-gitg/PKGBUILD]: opps! done.

**Comment by albfan** [comment]
I think this fails because libgit2-glib needs to be build with `--vapi=true`.

https://github.com/Alexpux/MINGW-packages/blob/master/mingw-w64-libgit2-glib/PKGBUILD#L36

libgit2-glib builds correctly with libgit2 0.27.0 but I think it does not work with 0.27.5 (actual libgit2)

Even more, there's a problem with libgit2-glib, where

https://gitlab.gnome.org/GNOME/libgit2-glib/blob/master/meson_vapi_link.py#L9

```
destdir = os.environ.get('DESTDIR', '')
dest_vapidir = os.path.normpath(destdir + os.sep + vapidir)
```

`destdir` is C:/msys64/mingw64, resulting `dest_vapidir` in something like:

    ../../pkg/mingw-w64-x86_64-libgit2-glib/C:/msys64/mingw64/share/vala/vapi

I can patch it and force `destdir` to be `/mingw64` (that's what I did locally), but don't know if there's a better option

Gonna start patching libgit2-glib to allow code review there

**Comment by albfan** [comment]
I need to bump libgit2-glib to 0.27.5 to make this work. I downgrade to libgit2 locally to 0.27.0 to make it work

**Comment by inigomartinez** [comment]
> destdir = os.environ.get('DESTDIR', '')
> dest_vapidir = os.path.normpath(destdir + os.sep + vapidir)
> 
> destdir is C:/msys64/mingw64, resulting dest_vapidir in something like:
> 
> ../../pkg/mingw-w64-x86_64-libgit2-glib/C:/msys64/mingw64/share/vala/vapi

Why is this? If `destdir` is `C:/msys64/mingw64` shouldn't it actually result in `C:/msys64/mingw64/share/vala/vapi`?

**Comment by inigomartinez** [comment]
Ok, I get it know, yes it would be wrong in Windows because `vapidir` is actually an absolute path, which works on *nix systems.

This can be fixed by using `get_option('datadir')` instead of `libgit2_glib_datadir` when [calling to meson_vapi_link.py](https://gitlab.gnome.org/GNOME/libgit2-glib/blob/master/meson.build#L135).

[meson_vapi_link.py](https://gitlab.gnome.org/GNOME/libgit2-glib/blob/master/meson_vapi_link.py#L9) should also use [MESON_INSTALL_DESTDIR_PREFIX]`http://mesonbuild.com/Reference-manual.html#meson-object`.

**Comment by albfan** [comment]
@Alexpux, no idea how to check appveyor problem

**Comment by albfan** [comment]
Seems Azure just ignores it https://dev.azure.com/msys2/mingw/_build/results?buildId=359&view=logs

**Comment by albfan** [comment]
Seems those "unexpected" messages are not errors and

    Command '['C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-introspectw4l4xtuy/Ggit-1.0.exe', '--introspect-dump=C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-introspectw4l4xtuy/functions.txt,C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-introspectw4l4xtuy/dump.xml']' returned non-zero exit status 3221225781.

means there's some error in Ggit-1.0.exe with an missing ddl (error 3221225781)

I see appveyor is deactivated by now

**Comment by inigomartinez** [comment]
Is there any `ldd` equivalent in Windows to check what library is missing?

**Comment by albfan** [comment]
@inigomartinez the problem is we should recreate the gnome.generate_gir task to look into that

**Comment by lazka** [comment]
I can't reproduce the build error locally (might be racy, or a appveyor issue), but some libraries are not found:

```
$ ntldd.exe gitg
        KERNEL32.dll => C:\Windows\system32\KERNEL32.dll (0x00800000)
        msvcrt.dll => C:\Windows\system32\msvcrt.dll (0x006f0000)
        libgdk-3-0.dll => C:\msys64\mingw32\bin\libgdk-3-0.dll (0x00800000)
        libgee-0.8-2.dll => C:\msys64\mingw32\bin\libgee-0.8-2.dll (0x006f0000)
        libgio-2.0-0.dll => C:\msys64\mingw32\bin\libgio-2.0-0.dll (0x00940000)
        libgirepository-1.0-1.dll => C:\msys64\mingw32\bin\libgirepository-1.0-1.dll (0x00330000)
        libgit2-glib-1.0-0.dll => C:\msys64\mingw32\bin\libgit2-glib-1.0-0.dll (0x00330000)
        libglib-2.0-0.dll => C:\msys64\mingw32\bin\libglib-2.0-0.dll (0x00e60000)
        libgobject-2.0-0.dll => C:\msys64\mingw32\bin\libgobject-2.0-0.dll (0x006f0000)
        libgtk-3-0.dll => C:\msys64\mingw32\bin\libgtk-3-0.dll (0x013e0000)
        libgtksourceview-3.0-1.dll => C:\msys64\mingw32\bin\libgtksourceview-3.0-1.dll (0x006f0000)
        libgtkspell3-3-0.dll => C:\msys64\mingw32\bin\libgtkspell3-3-0.dll (0x00030000)
        libintl-8.dll => C:\msys64\mingw32\bin\libintl-8.dll (0x00b50000)
        libpango-1.0-0.dll => C:\msys64\mingw32\bin\libpango-1.0-0.dll (0x00330000)
        libpeas-1.0-0.dll => C:\msys64\mingw32\bin\libpeas-1.0-0.dll (0x00330000)
        libgitg-ext-1.0-0.dll => not found
        libgitg-1.0-0.dll => not found
```
One solution would be to do this after ninja install:
```
mv "${pkgdir}${MINGW_PREFIX}"/lib/*.dll* "${pkgdir}${MINGW_PREFIX}"/bin
```

**Comment by inigomartinez** [comment]
The fact that `libgitg-ext-1.0-0.dll` and `libgitg-1.0-0.dll` seems the source of the issue.

I have been searching some kind of `LD_LIBRARY_PATH` for windows equivalent with no success, so it doesn't seem like a solution for this.

What about `glib` and `gtk`? both also support meson. If they are built using meson, what do they do with libraries? do they also move them after building them?

Are there any other modules that are also built with meson?

**Comment by lazka** [comment]
> I have been searching some kind of `LD_LIBRARY_PATH` for windows equivalent with no success, so it doesn't seem like a solution for this.

Windows searches in PATH instead.

> What about `glib` and `gtk`? both also support meson. If they are built using meson, what do they do with libraries? do they also move them after building them?

They don't install internal libraries. I don't know what's the right thing to do here... moving them should work for now at least.

> Are there any other modules that are also built with meson in windows? They could serve as a reference.

Here is a list under "Required By": https://msys2.duckdns.org/package/mingw-w64-i686-meson?repo=mingw32

**Comment by inigomartinez** [comment]
> They don't install internal libraries. I don't know what's the right thing to do here... moving them should work for now at least.

What about shared libraries?

```
libgio-2.0-0.dll => C:\msys64\mingw32\bin\libgio-2.0-0.dll (0x00940000)
libglib-2.0-0.dll => C:\msys64\mingw32\bin\libglib-2.0-0.dll (0x00e60000) libgobject-2.0-0.dll => C:\msys64\mingw32\bin\libgobject-2.0-0.dll (0x006f0000) libgtk-3-0.dll => C:\msys64\mingw32\bin\libgtk-3-0.dll (0x013e0000)
```

@albfan what about building the necessary libraries as static using meson options?

**Comment by albfan** [comment]
@lazka  @inigomartinez I get the same error on my system (windows 10) will try your suggestion to detect what's happening

**Comment by albfan** [comment]
After run locally:

    makepkg -sif

If you navigate to `MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32`

And run custom command for build `libgit2-glib/Ggit-1.0.gir` (extracted from build.ninja):

```
"C:/msys64/mingw64/bin/g-ir-scanner.EXE" -v "-pthread" "-mms-bitfields" "-IC:/msys64/mingw64/include/gobject-introspection-1.0" "-IC:/msys64/mingw64/include" "-IC:/msys64/mingw64/lib/libffi-3.2.1/include" "-IC:/msys64/mingw64/include/glib-2.0" "-IC:/msys64/mingw64/lib/glib-2.0/include" "-IC:/msys64/mingw64/include" "--no-libtool" "--namespace=Ggit" "--nsversion=1.0" "--warn-all" "--output" "libgit2-glib/Ggit-1.0.gir" "--c-include=libgit2-glib/ggit.h" "-IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/libgit2-glib-0.26.4/libgit2-glib" "-IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/libgit2-glib" "-I./." "-I../libgit2-glib-0.26.4/." "--filelist=C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/libgit2-glib/libgit2-glib@@git2-glib-1.0@sha/Ggit_1.0_gir_filelist" "--include=GObject-2.0" "--include=GLib-2.0" "--include=Gio-2.0" "--symbol-prefix=ggit" "--identifier-prefix=Ggit" "--pkg-export=libgit2-glib-1.0" "--cflags-begin" "-ffast-math" "-fstrict-aliasing" "-Wpointer-arith" "-Wmissing-declarations" "-Wformat=2" "-Wstrict-prototypes" "-Wmissing-prototypes" "-Wnested-externs" "-Wold-style-definition" "-Wdeclaration-after-statement" "-Wunused" "-Wuninitialized" "-Wshadow" "-Wmissing-noreturn" "-Wmissing-format-attribute" "-Wredundant-decls" "-Wlogical-op" "-Wcast-align" "-Wno-unused-local-typedefs" "-Werror=implicit" "-Werror=init-self" "-Werror=main" "-Werror=missing-braces" "-Werror=return-type" "-Werror=array-bounds" "-Werror=write-strings" "-DG_DISABLE_CAST_CHECKS" "-DG_DISABLE_CHECKS" "-I./." "-I../libgit2-glib-0.26.4/." "-IC:/msys64/mingw64/include" "-IC:/msys64/mingw64/lib/libffi-3.2.1/include" "-IC:/msys64/mingw64/include/glib-2.0" "-IC:/msys64/mingw64/lib/glib-2.0/include" "--cflags-end" "--library" "git2-glib-1.0" "-LC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/libgit2-glib" "-LC:/msys64/mingw64/lib" "--extra-library=gio-2.0" "--extra-library=gobject-2.0" "--extra-library=glib-2.0" "--extra-library=intl" "--extra-library=git2"
```

You will get the error:

```
C:\msys64\mingw64\bin/x86_64-w64-mingw32-gcc.exe -E -DG_DISABLE_CAST_CHECKS -DG_DISABLE_CHECKS -I. -IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32 -IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/libgit2-glib-0.26.4 -IC:/msys64/mingw64/include -IC:/msys64/mingw64/lib/libffi-3.2.1/include -IC:/msys64/mingw64/include/glib-2.0 -IC:/msys64/mingw64/lib/glib-2.0/include -IC:/msys64/mingw64/include/gobject-introspection-1.0 -IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/libgit2-glib-0.26.4/libgit2-glib -IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/libgit2-glib -o g-ir-cpp-yhh2347x.i -C -ffast-math -fstrict-aliasing -Wpointer-arith -Wmissing-declarations -Wformat=2 -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wdeclaration-after-statement -Wunused -Wuninitialized -Wshadow -Wmissing-noreturn -Wmissing-format-attribute -Wredundant-decls -Wlogical-op -Wcast-align -Wno-unused-local-typedefs -Werror=implicit -Werror=init-self -Werror=main -Werror=missing-braces -Werror=return-type -Werror=array-bounds -Werror=write-strings C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/g-ir-cpp-yhh2347x.c
C:\msys64\mingw64\bin/x86_64-w64-mingw32-gcc.exe -Wno-unused-result -Wsign-compare -DNDEBUG -march=x86-64 -mtune=generic -O2 -pipe -fwrapv -D__USE_MINGW_ANSI_STDIO=1 -D_WIN32_WINNT=0x0601 -DNDEBUG -march=x86-64 -mtune=generic -O2 -pipe -fwrapv -D__USE_MINGW_ANSI_STDIO=1 -D_WIN32_WINNT=0x0601 -DNDEBUG -DG_DISABLE_CAST_CHECKS -DG_DISABLE_CHECKS -IC:/msys64/mingw64/include/gobject-introspection-1.0 -IC:/msys64/mingw64/include -IC:/msys64/mingw64/lib/libffi-3.2.1/include -IC:/msys64/mingw64/include/glib-2.0 -IC:/msys64/mingw64/lib/glib-2.0/include -IC:/msys64/mingw64/include -IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/libgit2-glib-0.26.4/libgit2-glib -IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/libgit2-glib -IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32 -IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/libgit2-glib-0.26.4 -IC:/msys64/mingw64/include -IC:/msys64/mingw64/lib/libffi-3.2.1/include -IC:/msys64/mingw64/include/glib-2.0 -IC:/msys64/mingw64/lib/glib-2.0/include -IC:/msys64/mingw64/include -IC:/msys64/mingw64/include -IC:/msys64/mingw64/lib/libffi-3.2.1/include -IC:/msys64/mingw64/include -IC:/msys64/mingw64/include/glib-2.0 -IC:/msys64/mingw64/lib/glib-2.0/include -IC:/msys64/mingw64/include -IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32 -IC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/libgit2-glib-0.26.4 -IC:/msys64/mingw64/include -IC:/msys64/mingw64/lib/libffi-3.2.1/include -IC:/msys64/mingw64/include/glib-2.0 -IC:/msys64/mingw64/lib/glib-2.0/include -c C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-introspectx62h6u9b/Ggit-1.0.c -o C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-introspectx62h6u9b/Ggit-1.0.o -Wno-deprecated-declarations -pthread -mms-bitfields -ffast-math -fstrict-aliasing -Wpointer-arith -Wmissing-declarations -Wformat=2 -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wdeclaration-after-statement -Wunused -Wuninitialized -Wshadow -Wmissing-noreturn -Wmissing-format-attribute -Wredundant-decls -Wlogical-op -Wcast-align -Wno-unused-local-typedefs -Werror=implicit -Werror=init-self -Werror=main -Werror=missing-braces -Werror=return-type -Werror=array-bounds -Werror=write-strings
g-ir-scanner: link: cc -o C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-introspectx62h6u9b/Ggit-1.0.exe C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-introspectx62h6u9b/Ggit-1.0.o -lgit2-glib-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl-lgit2 -LC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/libgit2-glib -Wl,-rpath,C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/libgit2-glib -LC:/msys64/mingw64/lib -Wl,-rpath,C:/msys64/mingw64/lib -LC:/msys64/mingw64/lib -lgio-2.0 -lgobject-2.0 -lgmodule-2.0 -pthread -lglib-2.0 -lintl
Command '['C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-introspectx62h6u9b/Ggit-1.0.exe', '--introspect-dump=C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-introspectx62h6u9b/functions.txt,C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-introspectx62h6u9b/dump.xml']' returned non-zero exit status 3221225781.
' at ';'4\mingw64\x86_64-w64-mingw32\include\sec_api\string_s.h:42: syntax error, unexpected ';' in '  ;
' at ';'4\mingw64\x86_64-w64-mingw32\include\sec_api\string_s.h:69: syntax error, unexpected ';' in '  ;
' at ';'4\mingw64\x86_64-w64-mingw32\include\sec_api\string_s.h:71: syntax error, unexpected ';' in '  ;
```

After succesfully stop command at right moment I was able to keep relevant files: changing tmp-<whatever> for tmp-new

```
tmp-new
    ├── Ggit-1.0.c
    └── Ggit-1.0.o
```

And launch specific command to build Ggit-1.0.exe

```
cc -o C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-new/Ggit-1.0.exe C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-new/Ggit-1.0.o -lgit2-glib-1.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl -lgit2 -LC:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/libgit2-glib -Wl,-rpath,C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/libgit2-glib -LC:/msys64/mingw64/lib -Wl,-rpath,C:/msys64/mingw64/lib -LC:/msys64/mingw64/lib -lgio-2.0 -lgobject-2.0 -lgmodule-2.0 -pthread -lglib-2.0 -lintl
```

Once build I launch the next command:

```
'C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-new/Ggit-1.0.exe' '--introspect-dump=C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-new/functions.txt,C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-new/dump.xml'
```

getting the error:

```
C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-new/Ggit-1.0.exe: error while loading shared libraries: libgit2-glib-1.0-0.dll: cannot open shared object file: No such file or directory
```

If I go to libgit2-glib (where libgit2-glib-1.0-0.dll is) I get the error:

```
C:/Users/afanjul/projects/c/MINGW-packages/mingw-w64-libgit2-glib/src/build-x86_64-w64-mingw32/tmp-new
[tmp-new.zip](https://github.com/Alexpux/MINGW-packages/files/2498921/tmp-new.zip)
/Ggit-1.0.exe: error while loading shared libraries: ?: cannot open shared object file: No such file or directory
```

I have uploaded the tmp-new dir if you @lazka or @inigomartinez are able to check other things that can be failing here

**Comment by lazka** [comment]
fyi, you can set `GI_SCANNER_DEBUG=save-temps` for the g-i files to stay around after the build.

**Comment by albfan** [comment]
@Alexpux after some digging with strace I found that appveyor is looking for:

```
libssl-1_1.dll
libssl-1_1-x64.dll
libcrypto-1_1.dll
libcrypto-1_1-x64.dll
```

Seems it is not enough with mingw-w64-openssl. After add that libraries (download from anywhere on internet) build is succesful.

How should I fix this?, modifying something on mingw-w64-openssl?

**Comment by lazka** [comment]
Maybe related to https://github.com/Alexpux/MINGW-packages/issues/4458#issuecomment-425656385 then

**Comment by lazka** [comment]
But I guess that can't be it because it's failing for you too. openssl is a dependency of libgit2, so I don't understand how this can happen.

**Comment by albfan** [comment]
I think libgit2 needs to be recompiled:

Ggit-1.0.exe need libgit2 (-lgit2) and checking libgit2.dll

```
$ ntldd /c/msys64/mingw32/bin/libgit2.dll
        ADVAPI32.dll => C:\WINDOWS\SYSTEM32\ADVAPI32.dll (0x00e10000)
        KERNEL32.dll => C:\WINDOWS\SYSTEM32\KERNEL32.dll (0x00e90000)
        msvcrt.dll => C:\WINDOWS\SYSTEM32\msvcrt.dll (0x00e90000)
        USER32.dll => C:\WINDOWS\SYSTEM32\USER32.dll (0x00ee0000)
        WS2_32.dll => C:\WINDOWS\SYSTEM32\WS2_32.dll (0x00fa0000)
        libgcc_s_dw2-1.dll => C:\msys64\mingw32\bin\libgcc_s_dw2-1.dll (0x00660000)
        libcrypto-1_1.dll => not found
        libcurl-4.dll => C:\msys64\mingw32\bin\libcurl-4.dll (0x00e10000)
        libhttp_parser-2.dll => C:\msys64\mingw32\bin\libhttp_parser-2.dll (0x001a0000)
        libssh2-1.dll => C:\msys64\mingw32\bin\libssh2-1.dll (0x00e10000)
        libssl-1_1.dll => not found
        zlib1.dll => C:\msys64\mingw32\bin\zlib1.dll (0x00660000)
```

I added manually as said. I will trigger a new recompilation of libgit2 and see what happens

**Comment by albfan** [comment]
I think libgit2 needs to be recompiled:

Ggit-1.0.exe need libgit2 (-lgit2) and checking libgit2.dll

```
$ ntldd /c/msys64/mingw32/bin/libgit2.dll
        ADVAPI32.dll => C:\WINDOWS\SYSTEM32\ADVAPI32.dll (0x00e10000)
        KERNEL32.dll => C:\WINDOWS\SYSTEM32\KERNEL32.dll (0x00e90000)
        msvcrt.dll => C:\WINDOWS\SYSTEM32\msvcrt.dll (0x00e90000)
        USER32.dll => C:\WINDOWS\SYSTEM32\USER32.dll (0x00ee0000)
        WS2_32.dll => C:\WINDOWS\SYSTEM32\WS2_32.dll (0x00fa0000)
        libgcc_s_dw2-1.dll => C:\msys64\mingw32\bin\libgcc_s_dw2-1.dll (0x00660000)
        libcrypto-1_1.dll => not found
        libcurl-4.dll => C:\msys64\mingw32\bin\libcurl-4.dll (0x00e10000)
        libhttp_parser-2.dll => C:\msys64\mingw32\bin\libhttp_parser-2.dll (0x001a0000)
        libssh2-1.dll => C:\msys64\mingw32\bin\libssh2-1.dll (0x00e10000)
        libssl-1_1.dll => not found
        zlib1.dll => C:\msys64\mingw32\bin\zlib1.dll (0x00660000)
```

I added manually as said. I will trigger a new recompilation of libgit2 and see what happens

**Comment by albfan** [comment]
I think libgit2 needs to be recompiled:

Ggit-1.0.exe need libgit2 (-lgit2) and checking libgit2.dll

```
$ ntldd /c/msys64/mingw32/bin/libgit2.dll
        ADVAPI32.dll => C:\WINDOWS\SYSTEM32\ADVAPI32.dll (0x00e10000)
        KERNEL32.dll => C:\WINDOWS\SYSTEM32\KERNEL32.dll (0x00e90000)
        msvcrt.dll => C:\WINDOWS\SYSTEM32\msvcrt.dll (0x00e90000)
        USER32.dll => C:\WINDOWS\SYSTEM32\USER32.dll (0x00ee0000)
        WS2_32.dll => C:\WINDOWS\SYSTEM32\WS2_32.dll (0x00fa0000)
        libgcc_s_dw2-1.dll => C:\msys64\mingw32\bin\libgcc_s_dw2-1.dll (0x00660000)
        libcrypto-1_1.dll => not found
        libcurl-4.dll => C:\msys64\mingw32\bin\libcurl-4.dll (0x00e10000)
        libhttp_parser-2.dll => C:\msys64\mingw32\bin\libhttp_parser-2.dll (0x001a0000)
        libssh2-1.dll => C:\msys64\mingw32\bin\libssh2-1.dll (0x00e10000)
        libssl-1_1.dll => not found
        zlib1.dll => C:\msys64\mingw32\bin\zlib1.dll (0x00660000)
```

I added manually as said. I will trigger a new recompilation of libgit2 and see what happens

**Comment by albfan** [comment]
I found the culprit: libgit2 depends on libssh2 which depends on libcrypto-1_1.dll. I recompile libssh2 and now this build without libcrypto-1_1.dll avaliable.

Let's trigger a rebuild

**Pull request merged**